app/js/main.js

```diff
@@ -201,6 +211,39 @@ function setupMeteringDemo(subscriptions) {
     location.reload();
   });
 
+  // Determine whether there is a publisher based entitlement
+  let entitlement = null;
+  switch (getQueryParams().publisherEntitlement) {
+    case 'free':
+      entitlement = EntitlementType.FREE;
+      break;
+    case 'subscribed':
+      entitlement = EntitlementType.SUBSCRIBED;
+      break;
+    case 'metered':
+      entitlement = EntitlementType.METERED;
+      break;
+    default:
+      // See if someone set publisherEntitlement to one of the enums
+      entitlement = getQueryParams().publisherEntitlement || null;
+  }
+
+  if (
+    entitlement == EntitlementType.FREE ||
```

Comment:
Could you move these three cases to their own flows?

Reply:
I don't mind if you redo this part of code the way you want it.

Reply:
I'd still like to see that change in this PR

Reply:
I noticed you marked this one as resolved. I'm still seeing the three cases in the "?metering" flow though. Could you add them to one or more new flows instead? Just to clarify, when I say new flow(s) I mean like adding code that checks the flow (ex: `flow == 'metering') and then a custom function as well to match your refactoring in https://github.com/subscriptions-project/scenic-demo/pull/196 . Thanks!

Reply:
I'm not sure what you're trying to get at here.

Reply:
These cases are all part of metering

Reply:
That's a good point! I'll resolve this